### DISCUSSION TITLE
"Plex URL is invalid"

### DISCUSSION THREAD

**Discussion by michael-muller-tech** [meisnate12/Plex-Meta-Manager#2306] [General]
Hi, 

I read through all of the 'Plex URL is invalid' posts and could not find a solution sadly.

Below my config:

`libraries:                           # This is called out once within the config.yml file
  Films:                            # These are names of libraries in your Plex
    remove_overlays: false           # Set this to true to remove all overlays
    collection_files:
    - default: basic                 # This is a file within Kometa's defaults folder
    - default: imdb                  # This is a file within Kometa's defaults folder
    overlay_files:
    - default: ribbon                # This is a file within Kometa's defaults folder
  Series:
    remove_overlays: false           # Set this to true to remove all overlays
    collection_files:
    - default: basic                 # This is a file within Kometa's defaults folder
    - default: imdb                  # This is a file within Kometa's defaults folder
      # see the wiki for how to use local files, folders, URLs, or files from git
    overlay_files:
    - default: ribbon                # This is a file within Kometa's defaults folder
settings:
  run_order:
  - operations
  - metadata
  - collections
  - overlays
  cache: true
  cache_expiration: 60
  asset_directory: config/assets
  asset_folders: true
  asset_depth: 0
  create_asset_folders: false
  prioritize_assets: false
  dimensional_asset_rename: false
  download_url_assets: false
  show_missing_season_assets: false
  show_missing_episode_assets: false
  show_asset_not_needed: true
  sync_mode: append
  minimum_items: 1
  default_collection_order:
  delete_below_minimum: true
  delete_not_scheduled: false
  run_again_delay: 2
  missing_only_released: false
  only_filter_missing: false
  show_unmanaged: true
  show_unconfigured: true
  show_filtered: false
  show_options: true
  show_missing: true
  show_missing_assets: true
  save_report: false
  tvdb_language: eng
  ignore_ids:
  ignore_imdb_ids:
  item_refresh_delay: 0
  playlist_sync_to_users: all
  playlist_exclude_users:
  playlist_report: false
  verify_ssl: false
  custom_repo:
  overlay_artwork_filetype: jpg
  overlay_artwork_quality: 75
plex:
  url: http://192.168.178.10:32400/
  token: my-token
  timeout: 60
  db_cache:
  clean_bundles: false
  empty_trash: false
  optimize: false
  verify_ssl:
tmdb:                                # REQUIRED for the script to run
  apikey: my-api-key
  language: en
  cache_expiration: 60
  region:
#tautulli:                            # Can be individually specified per library as well
#  url: ##############http://192.168.1.12:8181
#  apikey: ################################
#github:
#  token: ################################
`

Command I run:

`
myname@myserver:~/Documents/Docker stuff/Kometa$ docker run --rm -it -v "/home/myname/Documents/Docker stuff/Kometa/config:/config:rw" kometateam/kometa --run

`
Gets me this error:

`
| Traceback (most recent call last):                                                                 |
|   File "/modules/config.py", line 1287, in __init__                                                |
|     raise Failed("Config Error: No libraries were found in config")                                |
| modules.util.Failed: Config Error: No libraries were found in config                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "//kometa.py", line 333, in start                                                           |
|     config = ConfigFile(my_requests, default_dir, attrs, secret_args)                              |
|              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                              |
|   File "/modules/config.py", line 1287, in __init__                                                |
|     raise Failed("Config Error: No libraries were found in config")                                |
| modules.util.Failed: Config Error: No libraries were found in config                               |
|                                                                                                    |
| Config Error: No libraries were found in config                                                    |
|                                                                                                    |
|========================================== Error Summary ===========================================|
|                                                                                                    |
| Count | Message                                                                                    |
|=======|============================================================================================|
|     2 | Plex Error: Plex URL is probably invalid                                                   |
|                                                                                                    |
|========================================= Critical Summary =========================================|
|                                                                                                    |
| Count | Message                                                                                    |
|=======|============================================================================================|
|     1 | Config Error: No libraries were found in config                                            |
|                                                                                                    |
|====================================================================================================|
|                                            Finished Run                                            |
|                                           Version: 2.1.0                                           |
|                          Finished: 22:36:55 2024-10-31 Run Time: 0:00:06                           |
|====================================================================================================|

`
The library names etc. are correct. 

Both docker containers are on the same (macvlan) network and I can curl the PleX container from my Kometa container. 
Connection type is not set to 'Secure' in PleX either.  Token/API verified multiple times too. 

Not sure what else I can try. Any ideas welcome. 
Thank you in advance.

**Comment by chazlarson** [comment]
Can you provide the full log?  The error you show there is the very last stage after Kometa could not connect to Plex, so there will be an error associated with the actual connection attempt further up.

You also may want to take this to the discord for more eyes.

**Reply by michael-muller-tech** [reply]
`
|====================================================================================================|
| Initializing cache database at //config/config.cache                                               |
|====================================================================================================|
| notifiarr attribute not found                                                                      |
| gotify attribute not found                                                                         |
|====================================================================================================|
| Connecting to TMDb...                                                                              |
| TMDb Connection Successful                                                                         |
|====================================================================================================|
| omdb attribute not found                                                                           |
|====================================================================================================|
| mdblist attribute not found                                                                        |
|====================================================================================================|
| trakt attribute not found                                                                          |
|====================================================================================================|
| mal attribute not found                                                                            |
|====================================================================================================|
| playlist_files attribute not found                                                                 |
|====================================================================================================|
| Connecting to Plex Libraries...                                                                    |
|====================================================================================================|
|                                        Films Configuration                                         |
|====================================================================================================|
|                                                                                                    |
| Connecting to Films Library...                                                                     |
|                                                                                                    |
| Reading in Collection Files                                                                        |
|                                                                                                    |
| Reading default: basic                                                                             |
|                                                                                                    |
| Reading default: imdb                                                                              |
|                                                                                                    |
| Reading in Overlay Files                                                                           |
|                                                                                                    |
| Reading default: ribbon                                                                            |
|                                                                                                    |
|======================================== Plex Configuration ========================================|
|                                                                                                    |
| Using Asset Directory: config/assets                                                               |
| Overriding verify_ssl to False for Plex connection                                                 |
| Plex Error: Plex connection attempt returned 'ConnectionError' or 'ParseError'                     |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 199, in _new_conn     |
|     sock = connection.create_connection(                                                           |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 85, in create_connection |
|     raise err                                                                                      |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 73, in create_connection |
|     sock.connect(sa)                                                                               |
| OSError: [Errno 113] No route to host                                                              |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 789, in urlopen   |
|     response = self._make_request(                                                                 |
|                ^^^^^^^^^^^^^^^^^^^                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 495, in _make_request |
|     conn.request(                                                                                  |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 441, in request       |
|     self.endheaders()                                                                              |
|   File "/usr/local/lib/python3.11/http/client.py", line 1281, in endheaders                        |
|     self._send_output(message_body, encode_chunked=encode_chunked)                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 1041, in _send_output                      |
|     self.send(msg)                                                                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 979, in send                               |
|     self.connect()                                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 279, in connect       |
|     self.sock = self._new_conn()                                                                   |
|                 ^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 214, in _new_conn     |
|     raise NewConnectionError(                                                                      |
| urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x791ad8540090>: Failed to establish a new connection: [Errno 113] No route to host |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in send           |
|     resp = conn.urlopen(                                                                           |
|            ^^^^^^^^^^^^^                                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 843, in urlopen   |
|     retries = retries.increment(                                                                   |
|               ^^^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 519, in increment     |
|     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]                  |
|     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                            |
| urllib3.exceptions.MaxRetryError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8540090>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| During handling of the above exception, another exception occurred:                                |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/modules/plex.py", line 462, in __init__                                                   |
|     self.PlexServer = PlexServer(baseurl=self.url, token=self.token, session=self.session, timeout=self.timeout) |
|                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 116, in __init__          |
|     data = self.query(self.key, timeout=self._timeout)                                             |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                             |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 760, in query             |
|     response = method(url, headers=headers, params=params, timeout=timeout, **kwargs)              |
|                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^              |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 602, in get            |
|     return self.request("GET", url, **kwargs)                                                      |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                      |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request        |
|     resp = self.send(prep, **send_kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "//kometa.py", line 223, in new_send                                                        |
|     return old_send(*send_args, **kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send           |
|     r = adapter.send(request, **kwargs)                                                            |
|         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                            |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 700, in send           |
|     raise ConnectionError(e, request=request)                                                      |
| requests.exceptions.ConnectionError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8540090>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 199, in _new_conn     |
|     sock = connection.create_connection(                                                           |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 85, in create_connection |
|     raise err                                                                                      |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 73, in create_connection |
|     sock.connect(sa)                                                                               |
| OSError: [Errno 113] No route to host                                                              |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 789, in urlopen   |
|     response = self._make_request(                                                                 |
|                ^^^^^^^^^^^^^^^^^^^                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 495, in _make_request |
|     conn.request(                                                                                  |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 441, in request       |
|     self.endheaders()                                                                              |
|   File "/usr/local/lib/python3.11/http/client.py", line 1281, in endheaders                        |
|     self._send_output(message_body, encode_chunked=encode_chunked)                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 1041, in _send_output                      |
|     self.send(msg)                                                                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 979, in send                               |
|     self.connect()                                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 279, in connect       |
|     self.sock = self._new_conn()                                                                   |
|                 ^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 214, in _new_conn     |
|     raise NewConnectionError(                                                                      |
| urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x791ad8540090>: Failed to establish a new connection: [Errno 113] No route to host |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in send           |
|     resp = conn.urlopen(                                                                           |
|            ^^^^^^^^^^^^^                                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 843, in urlopen   |
|     retries = retries.increment(                                                                   |
|               ^^^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 519, in increment     |
|     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]                  |
|     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                            |
| urllib3.exceptions.MaxRetryError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8540090>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| During handling of the above exception, another exception occurred:                                |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/modules/plex.py", line 462, in __init__                                                   |
|     self.PlexServer = PlexServer(baseurl=self.url, token=self.token, session=self.session, timeout=self.timeout) |
|                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 116, in __init__          |
|     data = self.query(self.key, timeout=self._timeout)                                             |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                             |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 760, in query             |
|     response = method(url, headers=headers, params=params, timeout=timeout, **kwargs)              |
|                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^              |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 602, in get            |
|     return self.request("GET", url, **kwargs)                                                      |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                      |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request        |
|     resp = self.send(prep, **send_kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "//kometa.py", line 223, in new_send                                                        |
|     return old_send(*send_args, **kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send           |
|     r = adapter.send(request, **kwargs)                                                            |
|         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                            |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 700, in send           |
|     raise ConnectionError(e, request=request)                                                      |
| requests.exceptions.ConnectionError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8540090>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| During handling of the above exception, another exception occurred:                                |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/modules/config.py", line 1179, in __init__                                                |
|     library = Plex(self, params)                                                                   |
|               ^^^^^^^^^^^^^^^^^^                                                                   |
|   File "/modules/plex.py", line 504, in __init__                                                   |
|     raise Failed("Plex Error: Plex URL is probably invalid")                                       |
| modules.util.Failed: Plex Error: Plex URL is probably invalid                                      |
|                                                                                                    |
| Plex Error: Plex URL is probably invalid                                                           |
|                                                                                                    |
| Films Library Connection Failed                                                                    |
|====================================================================================================|
|                                        Series Configuration                                        |
|====================================================================================================|
|                                                                                                    |
| Connecting to Series Library...                                                                    |
|                                                                                                    |
| Reading in Collection Files                                                                        |
|                                                                                                    |
| Reading default: basic                                                                             |
|                                                                                                    |
| Reading default: imdb                                                                              |
|                                                                                                    |
| Reading in Overlay Files                                                                           |
|                                                                                                    |
| Reading default: ribbon                                                                            |
|                                                                                                    |
|======================================== Plex Configuration ========================================|
|                                                                                                    |
| Using Asset Directory: config/assets                                                               |
| Overriding verify_ssl to False for Plex connection                                                 |
| Plex Error: Plex connection attempt returned 'ConnectionError' or 'ParseError'                     |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 199, in _new_conn     |
|     sock = connection.create_connection(                                                           |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 85, in create_connection |
|     raise err                                                                                      |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 73, in create_connection |
|     sock.connect(sa)                                                                               |
| OSError: [Errno 113] No route to host                                                              |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 789, in urlopen   |
|     response = self._make_request(                                                                 |
|                ^^^^^^^^^^^^^^^^^^^                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 495, in _make_request |
|     conn.request(                                                                                  |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 441, in request       |
|     self.endheaders()                                                                              |
|   File "/usr/local/lib/python3.11/http/client.py", line 1281, in endheaders                        |
|     self._send_output(message_body, encode_chunked=encode_chunked)                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 1041, in _send_output                      |
|     self.send(msg)                                                                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 979, in send                               |
|     self.connect()                                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 279, in connect       |
|     self.sock = self._new_conn()                                                                   |
|                 ^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 214, in _new_conn     |
|     raise NewConnectionError(                                                                      |
| urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x791ad8459d50>: Failed to establish a new connection: [Errno 113] No route to host |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in send           |
|     resp = conn.urlopen(                                                                           |
|            ^^^^^^^^^^^^^                                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 843, in urlopen   |
|     retries = retries.increment(                                                                   |
|               ^^^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 519, in increment     |
|     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]                  |
|     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                            |
| urllib3.exceptions.MaxRetryError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8459d50>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| During handling of the above exception, another exception occurred:                                |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/modules/plex.py", line 462, in __init__                                                   |
|     self.PlexServer = PlexServer(baseurl=self.url, token=self.token, session=self.session, timeout=self.timeout) |
|                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 116, in __init__          |
|     data = self.query(self.key, timeout=self._timeout)                                             |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                             |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 760, in query             |
|     response = method(url, headers=headers, params=params, timeout=timeout, **kwargs)              |
|                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^              |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 602, in get            |
|     return self.request("GET", url, **kwargs)                                                      |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                      |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request        |
|     resp = self.send(prep, **send_kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "//kometa.py", line 223, in new_send                                                        |
|     return old_send(*send_args, **kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send           |
|     r = adapter.send(request, **kwargs)                                                            |
|         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                            |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 700, in send           |
|     raise ConnectionError(e, request=request)                                                      |
| requests.exceptions.ConnectionError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8459d50>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 199, in _new_conn     |
|     sock = connection.create_connection(                                                           |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 85, in create_connection |
|     raise err                                                                                      |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/connection.py", line 73, in create_connection |
|     sock.connect(sa)                                                                               |
| OSError: [Errno 113] No route to host                                                              |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 789, in urlopen   |
|     response = self._make_request(                                                                 |
|                ^^^^^^^^^^^^^^^^^^^                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 495, in _make_request |
|     conn.request(                                                                                  |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 441, in request       |
|     self.endheaders()                                                                              |
|   File "/usr/local/lib/python3.11/http/client.py", line 1281, in endheaders                        |
|     self._send_output(message_body, encode_chunked=encode_chunked)                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 1041, in _send_output                      |
|     self.send(msg)                                                                                 |
|   File "/usr/local/lib/python3.11/http/client.py", line 979, in send                               |
|     self.connect()                                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 279, in connect       |
|     self.sock = self._new_conn()                                                                   |
|                 ^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connection.py", line 214, in _new_conn     |
|     raise NewConnectionError(                                                                      |
| urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x791ad8459d50>: Failed to establish a new connection: [Errno 113] No route to host |
|                                                                                                    |
| The above exception was the direct cause of the following exception:                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 667, in send           |
|     resp = conn.urlopen(                                                                           |
|            ^^^^^^^^^^^^^                                                                           |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/connectionpool.py", line 843, in urlopen   |
|     retries = retries.increment(                                                                   |
|               ^^^^^^^^^^^^^^^^^^                                                                   |
|   File "/usr/local/lib/python3.11/site-packages/urllib3/util/retry.py", line 519, in increment     |
|     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]                  |
|     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                            |
| urllib3.exceptions.MaxRetryError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8459d50>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| During handling of the above exception, another exception occurred:                                |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/modules/plex.py", line 462, in __init__                                                   |
|     self.PlexServer = PlexServer(baseurl=self.url, token=self.token, session=self.session, timeout=self.timeout) |
|                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 116, in __init__          |
|     data = self.query(self.key, timeout=self._timeout)                                             |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                             |
|   File "/usr/local/lib/python3.11/site-packages/plexapi/server.py", line 760, in query             |
|     response = method(url, headers=headers, params=params, timeout=timeout, **kwargs)              |
|                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^              |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 602, in get            |
|     return self.request("GET", url, **kwargs)                                                      |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                      |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 589, in request        |
|     resp = self.send(prep, **send_kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "//kometa.py", line 223, in new_send                                                        |
|     return old_send(*send_args, **kwargs)                                                          |
|            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                          |
|   File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 703, in send           |
|     r = adapter.send(request, **kwargs)                                                            |
|         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                            |
|   File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 700, in send           |
|     raise ConnectionError(e, request=request)                                                      |
| requests.exceptions.ConnectionError: HTTPConnectionPool(redacted): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x791ad8459d50>: Failed to establish a new connection: [Errno 113] No route to host')) |
|                                                                                                    |
| During handling of the above exception, another exception occurred:                                |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "/modules/config.py", line 1179, in __init__                                                |
|     library = Plex(self, params)                                                                   |
|               ^^^^^^^^^^^^^^^^^^                                                                   |
|   File "/modules/plex.py", line 504, in __init__                                                   |
|     raise Failed("Plex Error: Plex URL is probably invalid")                                       |
| modules.util.Failed: Plex Error: Plex URL is probably invalid                                      |
|                                                                                                    |
| Plex Error: Plex URL is probably invalid                                                           |
|                                                                                                    |
| Series Library Connection Failed                                                                   |
|====================================================================================================|
| Traceback (most recent call last):                                                                 |
|   File "/modules/config.py", line 1287, in __init__                                                |
|     raise Failed("Config Error: No libraries were found in config")                                |
| modules.util.Failed: Config Error: No libraries were found in config                               |
|                                                                                                    |
| Traceback (most recent call last):                                                                 |
|   File "//kometa.py", line 333, in start                                                           |
|     config = ConfigFile(my_requests, default_dir, attrs, secret_args)                              |
|              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                              |
|   File "/modules/config.py", line 1287, in __init__                                                |
|     raise Failed("Config Error: No libraries were found in config")                                |
| modules.util.Failed: Config Error: No libraries were found in config                               |
|                                                                                                    |
| Config Error: No libraries were found in config                                                    |
|                                                                                                    |
|========================================== Error Summary ===========================================|
|                                                                                                    |
| Count | Message                                                                                    |
|=======|============================================================================================|
|     2 | Plex Error: Plex URL is probably invalid                                                   |
|                                                                                                    |
|========================================= Critical Summary =========================================|
|                                                                                                    |
| Count | Message                                                                                    |
|=======|============================================================================================|
|     1 | Config Error: No libraries were found in config                                            |
|                                                                                                    |
|====================================================================================================|
|                                            Finished Run                                            |
|                                           Version: 2.1.0                                           |
|                          Finished: 23:02:56 2024-10-31 Run Time: 0:00:07                           |
|====================================================================================================|

`

**Reply by chazlarson** [reply]
This container cannot see or access the IP in your Plex URL:
```
Failed to establish a new connection: [Errno 113] No route to host
```
That is probably because it's an ephemeral container on a different docker network.

**Reply by michael-muller-tech** [reply]
Yes, this is starting to make sense. 

Doing it from within the existing one gives me this: 

michael@servername:~/Documents/Docker stuff/Kometa$ docker exec -it pensive_saha kometa --run
OCI runtime exec failed: exec failed: unable to start container process: exec: "kometa": executable file not found in $PATH: unknown

I shall investigate further, but yeah, this is quite a big one to miss. Thanks so much.

**Reply by chazlarson** [reply]
That command should be:
```
docker exec -it pensive_saha python kometa.py --run
```

**Reply by michael-muller-tech** [reply]
Thank you! Somehow that's not in my path. I will try to find out why tomorrow. You have been extremely helpful.

**Reply by chazlarson** [reply]
~~Is this the official image?  If not, the files are likely elsewhere inside the container.~~

**Comment by chazlarson** [comment]
Also, that `docker run` command is creating and tearing down a new container with each run, not on the same named network, so whatever container it is from which you can curl the plex server, it is not the one created by that `docker run` command [which exists only while the script runs and fails].

If you have another Kometa container up and running, you probably want to exec into it and run the script inside *that* container, or put this ephemeral container on the same network.

**Reply by michael-muller-tech** [reply]
Thank you! This is likely what's not working here. Bit of an embarrassing error, but we move! Thanks so much. I will play around with the existing one instead and see whether I can work it out from there.
